obspy/clients/filesystem/sds.py

```diff
@@ -169,24 +224,274 @@ def get_waveforms(self, network, station, location, channel, starttime,
+                network=network, station=station, location=location,
+                channel=channel, time=time, sds_type=sds_type)
+            if os.path.isfile(filename):
+                st = read(filename, format=self.format, headonly=True)
+                if st:
+                    break
+            time -= 24 * 3600
+
+        return max([tr.stats.endtime for tr in st])
+
+    def get_latency(self, network, station, location, channel,
+                    sds_type=None, stop_time=None):
+        """
+        Get time of last sample for given stream.
+
+        :type network: str
+        :param network: Network code of requested data (e.g. "IU").
+        :type station: str
+        :param station: Station code of requested data (e.g. "ANMO").
+        :type location: str
+        :param location: Location code of requested data (e.g. "").
+        :type channel: str
+        :param channel: Channel code of requested data (e.g. "HHZ").
+        :type sds_type: str
+        :param sds_type: Override SDS data type identifier that was specified
+            during client initialization.
+        :type stop_time: :class:`~obspy.core.utcdatetime.UTCDateTime`
+        :param stop_time: Time at which the search for data is stopped and
+            `None` is returned. If not specified, stops at ``1950-01-01T00``.
+        :rtype: float or ``None``
+        :returns: Latency in seconds or ``None`` if no data was encountered
+            from current time backwards until ``stop_time``.
+        """
+        endtime = self._get_current_endtime(
+            network=network, station=station, location=location,
+            channel=channel, sds_type=sds_type, stop_time=stop_time)
+
+        if endtime is None:
+            return endtime
+
+        return UTCDateTime() - endtime
+
+    def has_data(self, network, station, location, channel, sds_type=None):
+        """
+        Check if specified stream has any data.
+
+        Actually just checks whether a file is encountered in a folder that is
+        expected to contain data.
+
+        :type network: str
+        :param network: Network code of requested data (e.g. "IU").
+        :type station: str
+        :param station: Station code of requested data (e.g. "ANMO").
+        :type location: str
+        :param location: Location code of requested data (e.g. "").
+        :type channel: str
+        :param channel: Channel code of requested data (e.g. "HHZ").
+        :type sds_type: str
+        :param sds_type: Override SDS data type identifier that was specified
+            during client initialization.
+        :rtype: bool
+        """
+        sds_type = sds_type or self.sds_type
+        pattern = self.FMTSTR.format(
+            year="*", network=network, station=station, location=location,
+            channel=channel, doy=0, type=sds_type)
+        # can not insert wildcard for day-of-year above, so replace it now
+        pattern = pattern.rsplit(".", 1)[0] + "*"
```

Comment:
Hmm, you're right, that's vulnerable.. I'll try to change this, replacing everything not in NSLC with `*`.